src/app/destinyTrackerApi/d2-itemListBuilder.ts

```diff
@@ -49,20 +49,24 @@ export function getVendorItemList(
 
 function getNewItems(allItems: D2Item[], reviewDataCache: D2ReviewDataCache) {
   const allDtrItems = allItems.map(translateToDtrItem);
-  const allKnownDtrItems = reviewDataCache.getItemStores();
+  const allKnownDtrItems = new Set(
+    reviewDataCache.getItemStores().map((kdi) => `${kdi.referenceId}-${kdi.roll}`)
+  );
 
-  const unmatched = allDtrItems.filter((di) =>
-    allKnownDtrItems.every((kdi) => di.referenceId !== kdi.referenceId)
+  const unmatched = allDtrItems.filter(
+    (di) => !allKnownDtrItems.has(`${di.referenceId}-${getD2Roll(di.availablePerks)}`)
   );
 
   return unmatched;
 }
 
 function getNewVendorItems(vendorItems: D2ItemFetchRequest[], reviewDataCache: D2ReviewDataCache) {
-  const allKnownDtrItems = reviewDataCache.getItemStores();
+  const allKnownDtrItems = new Set(
+    reviewDataCache.getItemStores().map((kdi) => `${kdi.referenceId}-${kdi.roll}`)
+  );
 
-  const unmatched = vendorItems.filter((vi) =>
-    allKnownDtrItems.every((di) => di.referenceId !== vi.referenceId)
+  const unmatched = vendorItems.filter(
+    (di) => !allKnownDtrItems.has(`${di.referenceId}-${getD2Roll(di.availablePerks)}`)
```

Comment:
We build this string independently here and in the reducer.

Probably makes sense to make a single function that computes + returns the string alongside `getD2Roll`.